### DISCUSSION TITLE
Snippet: How to lock blocks and prevent your users from unlocking them

### DISCUSSION THREAD

**Discussion by ndiego** [WordPress/developer-blog-content#342] [Topic Ideas]
This snippet would provide a very brief overview of the [Block Locking API](https://developer.wordpress.org/block-editor/how-to-guides/curating-the-editor-experience/block-locking/) and then show developers how to prevent unlocking in specific contexts: 

**Post types:**
```php
add_filter( 'block_editor_settings_all', function( $settings, $context ) {
    if ( $context->post && 'page' === $context->post->post_type ) {
        $settings['canLockBlocks'] = false;
    }

    return $settings;
}, 10, 2 );
```

**User permissions:**
```php
add_filter( 'block_editor_settings_all', function( $settings ) {
    $settings['canLockBlocks'] = current_user_can( 'edit_theme_options' );

    return $settings;
} );
```

The snippet will also show how to disable access to the Code Editor, so users cannot manipulate the attributes directly.

**Comment by bph** [comment]
This topic was approved and conversations continues on the issue
- #344

**Comment by ndiego** [comment]
The snippet has been published: https://developer.wordpress.org/news/snippets/how-to-lock-wordpress-blocks-and-prevent-unlocking/
